encode_datasets.py

```diff
@@ -2059,7 +2064,228 @@ def process(inputfile, outfile):
+    fmeta = open(f"{dataset}/{kind}_meta.txt", "w+")
+    counter=0
+    
+    if kind=='test':
+        df=pd.read_json("/content/cosmosqa/data/test.jsonl",lines=True)
+        questions=df[['id','context','question','answer0','answer1','answer2','answer3']].values
+    else:
+        df=pd.read_csv("/content/cosmosqa/data/"+file)
+        questions=df[['id','context','question','answer0','answer1','answer2','answer3','label']].values
+
+    for row in range(len(questions)):
+        id=questions[row][0]
+        contexts=questions[row][1].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        question=questions[row][2].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        if '?' not in question:
+            question = question + "?"
+        option1=" (A) "+questions[row][3].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        option2=" (B) "+questions[row][4].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        option3=" (C) "+questions[row][5].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        option4=" (D) "+questions[row][6].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        options = option1 + option2 + option3 + option4 
+
+        if kind=='test':
+            answer="-"
+            answer_string="-"
+        else:
+            answer_index = questions[row][7]
+            answer=chr(ord('A')+answer_index)
+            answer_string=questions[row][answer_index+3].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        fmeta.write(f"{id}\t{answer} \n")
+        fout.write(f"{question} \\n{options} \\n {contexts}\t{answer_string} \n")
+    return len(questions)
+
+def cosmosqa():
+    train_count=cosmosqa_process("train.csv","cosmos","train")
+    val_count=cosmosqa_process("valid.csv","cosmos","val")
+    test_count=cosmosqa_process("test.jsonl","cosmos","test")
+    with open(f"/content/cosmos/counts.json", "w+") as outfile:
+        json.dump({"train": train_count, "val": val_count, "test": test_count}, outfile)
         
+def tweetqa_process(file,dataset, kind):
+    fout = open(f"{dataset}/{kind}.tsv", "w+")
+    fmeta = open(f"{dataset}/{kind}_meta.txt", "w+")
+    df=pd.read_json("/content/"+file)
+
+    if kind=='test':
+        questions=df[['Question','Tweet','qid']].values
+    else:
+        questions=df[['Question','Tweet','qid','Answer']].values
+
+    for row in range(len(questions)):
+        id=questions[row][2]
+        contexts=questions[row][1].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        question=questions[row][0].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        if '?' not in question:
+            question = question + "?"
+        fmeta.write(f"{id}\n")
```

Comment:
Could you also dump all the correct answers here on the 2nd column of the meta-file? They could be separated with a special character like `///`.

Reply:
We have used `\t` for yes-no question meta-files. Is it okay to use it here as well?